sbt-utils.js

```diff
@@ -163,3 +213,102 @@ const sbtPluginsPath = function (projectPath) {
   return [];
 };
 exports.parseKVDep = parseKVDep;
+
+const sbtInvoker = function (debugMode, path, tempSbtPlugins) {
+  let SBT_CMD = process.env.SBT_CMD || "sbt";
+  let sbtVersion = determineSbtVersion(path);
+  if (debugMode) {
+    console.log("Detected sbt version: " + sbtVersion);
+  }
+  const standalonePluginFileSupported = standalonePluginFile(sbtVersion);
+
+  let enablePluginString =  isDependencyTreeBuiltIn(sbtVersion) ? ENABLE_SBT_PLUGIN : ADD_SBT_PLUGIN;
+
+  if (standalonePluginFileSupported) {
+    fs.writeFileSync(tempSbtPlugins, enablePluginString);
+  }
+
+  return {
+    invokeDependencyList: function(commandPrefix, basePath, timeoutMs) {
+      let dependencyListCmd;
+      
+      let tmpDir = pathLib.join(os.tmpdir(), 'cdxgen-sbt-');
+      const outDir = fs.mkdtempSync(tmpDir);
+
+      if (commandPrefix !== '') {
+        dependencyListCmd = `"${commandPrefix}customShiftLeftListDependencies ${outDir}"`;
+      } else {
+        dependencyListCmd = `"customShiftLeftListDependencies ${outDir}"`;
+      }
+
+      let pluginFileName = '';
+      let customPluginContent = isDependencyTreeBuiltIn(sbtVersion) ? pluginStringForBuiltin : pluginStringForNonBuiltin;
+      let customPluginFileName = addCustomPlugin(basePath, customPluginContent);
+      if (standalonePluginFileSupported) {
+        sbtArgs = [`-addPluginSbtFile=${tempSbtPlugins}`, dependencyListCmd];
+      } else {
+        // write to a new shiftleft-cdxgen-plugins0.sbt file
+        pluginFileName = addPlugin(basePath, enablePluginString);
+        sbtArgs = [dependencyListCmd];
+      }
+
+      try {
+        console.log(`Executing ${SBT_CMD} ${sbtArgs} in ${basePath}`)
+        // Note that the command has to be invoked with `shell: true` to properly execut sbt
+        const result = spawnSync(
+          SBT_CMD,
+          sbtArgs,
+          { cwd: basePath, shell: true, encoding: "utf-8", timeout: timeoutMs }
+        );
+        if (result.status == 1 || result.error) {
+          console.error(result.stdout, result.stderr);
+          utils.debug(`1. Check if scala and sbt is installed and available in PATH. Only scala 2.10 + sbt 0.13.6+ and 2.12 + sbt 1.0+ is supported for now.`);
+          utils.debug(`2. Check if the plugin net.virtual-void:sbt-dependency-graph 0.10.0-RC1 can be used in the environment`);
+        } else {
+          utils.debug(result.stdout);
+        }
+      } finally {
+        if (pluginFileName != '' && fs.existsSync(pluginFileName)) {
+          fs.rmSync(pluginFileName);
```

Comment:
Only available since `v14.14.0` and I'm on `v14.13.0` ... sigh. But seems like you can't make `unlink` synchronous either, so really need this unless rewriting everything else was an option. That's a bit annoying.